fip-0039.md

```diff
@@ -0,0 +1,155 @@
+|Invalid encrypt public key|Specified public key is not valid FIO format.|400|"encrypt_public_key"|Value sent in, e.g. "notakey"|"Encrypt key not a valid FIO Public Key."|
+|FIO Domain expired|Domain of supplied FIO Address has expired more than 30 days ago|400|"fio_address"|Value sent in, i.e. "purse@alice"|"FIO Domain expired."|
+|Invalid fee value|max_fee format is not valid|400|"max_fee"|Value sent in, e.g. "-100"|"Invalid fee value"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|"max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+|Insufficient balance|Available (unlocked and unstaked) balance in account is less than chain fee|400|"max_fee"|Value sent in, e.g. "100000000000"|"Insufficient balance"|
+|Invalid TPID|tpid format is not valid|400|"tpid"|Value sent in, e.g. "notvalidfioaddress"|"TPID must be empty or valid FIO address"|
+|Not owner of FIO Address|The signer does not own the FIO Address|403|||Type: invalid_signature|
+|Signer not actor|Signer not actor|403|||Type: invalid_signature|
+#### Response body
+|Parameter|Format|Definition|
+|---|---|---|
+|status|String|OK if successful|
+|fee_collected|Int|Amount of SUFs collected as fee|
+##### Example
+```
+{
+  "status": "OK",
+  "fee_collected": 0
+}
+```
+
+## New getters
+### Get encryption key
+Returns the FIO Public Key to use for encryption of New Funds Request/Record OBT content blob.
+#### New end point: */get_encrypt_key* 
+#### Request body
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_address|Yes|FIO Public Key|FIO Address to fetch encrypt key for.|
+##### Example
+```
+{
+  "fio_address": "purse@alice"
+}
+```
+##### Processing
+* Request is validated per Exception handling
+* Return encrypt_public_key set using *updcryptkey*. If not set return FIO Public Key mapped using add_pub_address for chain_code: FIO, token_code: FIO
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid FIO Address|Format of FIO Address not valid or FIO Address does not exist.|400|"fio_address"|Value sent in, i.e. "purse@alice"|"FIO Address invalid or does not exist."|
+#### Response body
+|Parameter|Format|Definition|
+|---|---|---|
+|encrypt_public_key|Int|FIO Public key to use for encryption.|
+##### Example
+```
+{
+  "encrypt_public_key": "FIO5G2tfhPZtiaiBU6o9SpT2XKbb28gtKUeXXarvBjxuU29VYgYCs"
+}
+```
+
+# Rationale
+The following alternative approaches were considered:
+* Enabling of _newaccount_ action. It is a system action now, but could have been made public. However, that approach was deemed more complex as it would require updates to other smart contracts which are calling this action now.
+* Storing the encryption key as a new _token_code_ for _chain_code:FIO_ using _add_pub_address_. However, it would have been harder to support backwards compatibility where the original key is returned when a user has not specified a new encryption key.
```

Comment:
I do not see how backward compatibility changes. Either way you need to check first in the new encrypt field and, if it is not set, then use the original FIO Pub Key.

Reply:
If you are thinking of adding a new field to the fionames table (?) that is not part of the addresses array list, I believe you will need to do a table migration which is a big effort. Would need to confirm with Casey.

Reply:
Please avoid table migrations if at all possible.